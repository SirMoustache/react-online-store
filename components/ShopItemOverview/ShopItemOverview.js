/**
 * Absolute imports
 */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Theme
 */
import { SHADOW_DEFAULT } from '../../theme/shadows';

const Root = styled.div`
  padding: 16px;
  margin: 16px;
  box-shadow: ${SHADOW_DEFAULT};
`;

const ShopItemOverview = ({ title }) => (
  <Root>
    <h3>Shop Item</h3>
    Title: {title}
  </Root>
);

ShopItemOverview.propTypes = {
  title: PropTypes.string,
};

export default ShopItemOverview;
