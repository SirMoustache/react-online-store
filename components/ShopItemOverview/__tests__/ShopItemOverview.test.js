/**
 * Absolute imports
 */
import React from 'react';
import { render, cleanup } from 'react-testing-library';

/**
 * Local Components
 */
import ShopItemOverview from '..';

const renderComponent = ({ ...rest } = {}) =>
  render(<ShopItemOverview {...rest} />);

describe('<ShopItemOverview/>', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
