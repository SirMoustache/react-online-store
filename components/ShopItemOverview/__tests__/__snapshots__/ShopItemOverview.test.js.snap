// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShopItemOverview/> should match snapshot 1`] = `
<div>
  <div
    class="ShopItemOverview__Root-sc-2bbkh4-0 hiZFRu"
  >
    <h3>
      Shop Item
    </h3>
    Title: 
  </div>
</div>
`;
